test/unit/resource_cache_test.rb

```diff
@@ -78,6 +78,15 @@ def test_concurrently_syncing_huge_numbers_of_resources_makes_exactly_one_kubect
     assert(all_resources.all?(&:synced?))
   end
 
+  def test_warming_the_resource_cache_means_no_more_kubectl_calls_after
+    deployments = build_fake_deployments(10) # these also get pods
+    pods = build_fake_pods(10)
+
+    stub_kind_get("FakeDeployment", items: deployments.map(&:kubectl_response), times: 1)
+    stub_kind_get("FakePod", items: pods.map(&:kubectl_response), times: 1)
+    @cache.prewarm(deployments) # Fetches both Deployments and Pods
```

Comment:
Don't you need to call `sync` after to prove no more calls are made? It seems to me that right now this test only proves that prewaming makes the calls, not that sync doesn't also make them.

Reply:
Yeah I originally had that in and it worked fine. The issue I encountered then is that deleting the prewarm line didn't make the test fail so it kind of like... didn't feel right.
Thoughts?

Reply:
> deleting the prewarm line didn't make the test fail 

I see what you mean... if `prewarm` was just `return true` it would pass. Maybe we need two tests: this one as-is, and another proving that the pre-warming gets used by subsequent `sync` calls?

> Would that be a better option here?

No need to split it across threads (I think this is really fast with the dummy resources), but that's more or less what I was trying to suggest. 😄